File: day6.js
const fs = require("fs");

function countTotalsInMap(totalPeople, answerMap) {
  let count = 0;
  answerMap.forEach((value) => {
    if (value === totalPeople) {
      count++;
    }
  });
  return count;
}

const input = fs.readFileSync("./input/day6.txt", "utf8").split("\r\n");

let totalCount = 0;
let allInGroupCount = 0;
let lineCount = 0;

const groupCountMap = new Map();

input.forEach((line) => {
  if (line === "") {
    totalCount += groupCountMap.size;
    allInGroupCount += countTotalsInMap(lineCount, groupCountMap);
    groupCountMap.clear();
    lineCount = 0;
    return;
  }
  lineCount++;
  for (let char of line) {
    if (!groupCountMap.has(char)) {
      groupCountMap.set(char, 1);
    } else {
      groupCountMap.set(char, groupCountMap.get(char) + 1);
    }
  }
});
//last line is a character string, not an empty string, so add the counts again
totalCount += groupCountMap.size;
allInGroupCount += countTotalsInMap(lineCount, groupCountMap);
console.log("Part 1 :", totalCount);
console.log("Part 2 :", allInGroupCount);

File: 2020/day6.py
print('Python')
file = open('./input/day6.txt')
file = [x.strip() for x in file.readlines()]

groupCountMap = {}


def countTotalsInMap(totalPeople, answerMap):
    count = 0
    for val in answerMap.values():
        if val == totalPeople:
            count += 1
    return count


totalCount = 0
allInGroupCount = 0
lineCount = 0
for line in file:
    if line == "":
        totalCount += len(groupCountMap)
        allInGroupCount += countTotalsInMap(lineCount, groupCountMap)
        groupCountMap.clear()
        lineCount = 0
        continue

    lineCount += 1
    for char in line:
        if char not in groupCountMap:
            groupCountMap[char] = 1
        else:
            groupCountMap[char] += 1

totalCount += len(groupCountMap)
allInGroupCount += countTotalsInMap(lineCount, groupCountMap)
print("Part 1 : {}".format(totalCount))
print("Part 2 : {}".format(allInGroupCount))